### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by pahill**

### CODE REVIEW

File: androidApp/build.gradle.kts [hunk 1]
```diff
@@ -29,18 +29,17 @@ android {
     }
 
     buildTypes {
-        create("debugPG") {
```
Comment (terrakok) [inline androidApp/build.gradle.kts:32]: why did you remove the custom build type? it was for a test proguard build

File: androidApp/build.gradle.kts [hunk 2]
```diff
@@ -60,11 +59,11 @@ android {
     compileOptions {
         // Flag to enable support for the new language APIs
         isCoreLibraryDesugaringEnabled = true
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
```
Comment (terrakok) [inline androidApp/build.gradle.kts:63]: why did you update java version? it is required for old android support

File: gradle.properties [hunk 3]
```diff
@@ -14,4 +14,5 @@ kotlin.mpp.androidSourceSetLayoutVersion=2
 android.useAndroidX=true
 android.compileSdk=33
 android.targetSdk=33
-android.minSdk=21
\ No newline at end of file
+android.minSdk=21
+android.defaults.buildfeatures.buildconfig=true
```
Comment (terrakok) [inline gradle.properties:18]: could you explain the flag?

File: gradle/libs.versions.toml
Comment (terrakok) [inline gradle/libs.versions.toml:1]: Is it possible to upgrade versions only? not move declarations

File: shared/build.gradle.kts [hunk 4]
```diff
@@ -5,10 +5,12 @@ plugins {
 }
 
 kotlin {
+    targetHierarchy.default()
+
     android {
         compilations.all {
             kotlinOptions {
-                jvmTarget = "1.8"
+                jvmTarget = "17"
```
Comment (terrakok) [inline shared/build.gradle.kts:13]: jvmTarget = "1.8" is required for android

File: shared/build.gradle.kts [hunk 5]
```diff
@@ -60,7 +62,7 @@ kotlin {
         val iosX64Main by getting
         val iosArm64Main by getting
         val iosSimulatorArm64Main by getting
-        val iosMain by creating {
+        val iosMain by getting {
             dependsOn(commonMain)
             iosX64Main.dependsOn(this)
```
Comment (terrakok) [inline shared/build.gradle.kts:67]: if we use the default hierarchy, then it is not required to have manual dependsOn edges

File: androidApp/build.gradle.kts (hunk 1 above)
Comment (pahill) [reply androidApp/build.gradle.kts:32]: Ah, is that what it was!

File: androidApp/build.gradle.kts (hunk 2 above)
Comment (pahill) [reply androidApp/build.gradle.kts]: Ah! I got an IDE warning. Have reverted, but I think soon we could update to 11?

File: gradle.properties (hunk 3 above)
Comment (pahill) [reply gradle.properties:18]: For some reason, BuildConfig was not being created or picked up. This flag helped.

File: shared/build.gradle.kts (hunk 4 above)
Comment (pahill) [reply shared/build.gradle.kts]: Got it, reverted

File: shared/build.gradle.kts (hunk 5 above)
Comment (pahill) [reply shared/build.gradle.kts]: Thank you

File: gradle/libs.versions.toml
Comment (pahill) [reply gradle/libs.versions.toml:1]: Sure! The upgrade plugins I used did this.

File: gradle.properties (hunk 3 above)
Comment (terrakok) [reply gradle.properties:18]: https://stackoverflow.com/a/74692390

File: shared/build.gradle.kts [hunk 6]
```diff
@@ -60,11 +62,7 @@ kotlin {
         val iosX64Main by getting
         val iosArm64Main by getting
         val iosSimulatorArm64Main by getting
```
Comment (terrakok) [inline shared/build.gradle.kts:64]: we don't need it anymore

File: shared/build.gradle.kts [hunk 7]
```diff
@@ -80,13 +78,10 @@ android {
 
     defaultConfig {
         minSdk = (findProperty("android.minSdk") as String).toInt()
-        targetSdk = (findProperty("android.targetSdk") as String).toInt()
     }
     compileOptions {
         // Flag to enable support for the new language APIs
         isCoreLibraryDesugaringEnabled = true
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (terrakok) [inline shared/build.gradle.kts:89]: are you sure that the compatibility in the app config is enough?

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -1,23 +1,23 @@
...
+plugin-kotlin = "1.8.21"
+plugin-gver = "0.47.0"
 
-androidx-compose-compiler = "1.4.3"
-androidx-compose = "1.3.1"
-androidx-compose-ui = "1.3.3"
-kotlinx-serialization = "1.5.0"
-kotlinx-coroutines = "1.6.4"
-ktor = "2.2.4"
+androidx-compose-compiler="1.4.7"
```
Comment (terrakok) [inline gradle/libs.versions.toml:6]: " = "

**Review by terrakok — CHANGES_REQUESTED** [review]

File: shared/build.gradle.kts (hunk 6 above)
Comment (pahill) [reply shared/build.gradle.kts:64]: Can you be more specific? Do you mean iosX64Main -> iosSimulatorArm64Main?

File: shared/build.gradle.kts (hunk 7 above)
Comment (pahill) [reply shared/build.gradle.kts:89]: I'll roll it back

File: gradle/libs.versions.toml (hunk 8 above)
Comment (pahill) [reply gradle/libs.versions.toml:6]: Got it

File: gradle.properties (hunk 3 above)
Comment (pahill) [reply gradle.properties:18]: Thanks

File: androidApp/build.gradle.kts [hunk 9]
```diff
@@ -32,15 +32,15 @@ android {
         create("debugPG") {
             isDebuggable = false
             isMinifyEnabled = true
-            versionNameSuffix = " debugPG"
```
[the remark this answers is not in the document]
Comment (pahill) [reply androidApp/build.gradle.kts:35]: No, thanks - corrected it

[the remark this answers is not in the document]
Comment (terrakok) [reply androidApp/build.gradle.kts:35]: 🤔 You corrected a wrong line

File: shared/build.gradle.kts (hunk 6 above)
Comment (terrakok) [reply shared/build.gradle.kts:64]: I mean, after removing the manual setup of the iosMain source set, we don't need these properties anymore

**Review by terrakok — CHANGES_REQUESTED** [review]

**Comment by terrakok** [comment]
After finalizing the PR, don't forget to build and run the apps on ios and android to verify it in a runtime 😉

**Comment by pahill** [comment]
> After finalizing the PR, don't forget to build and run the apps on ios and android to verify it in a runtime 😉

I always do!
